### DISCUSSION TITLE
Require reporters to select the rule which the reported flip is breaking

### DISCUSSION THREAD

**Discussion by Noobmasser** [idena-network/idena-docs#133] [IIPs]
### Existing problems:

- Validators are overspending reports because it provides rewards
- Identities that got their flip reported are unable to learn from their mistakes and do better

### Proposal:

**Require reporters to select the rule which the reported flip is breaking.**

As we have clearly defined rules that hardly allow overstepping, it is necessary for the Validators to follow the rules instead of their gut feeling about the flip. When tasked with providing reason, the Reporter would appraise the flip more carefully.

In the explorer anyone should be able to see not only the report status, but the broken rule as well, which would allow the creator to improve and learn on their mistakes instead of being mad at the system or blaming random factors.

Later additional systems can be derived from this feature if necessary, like ignoring the report altogether if the reporters can't agree on the rule that the reported flip is breaking (please take note that this example is not a part of the proposal).

The community and developers would also be able to see current issues more clearly through gathering the data, being able to identify, for example, influx of "waking up" templates or "screen templates".

**UI:**

Upon clicking "report" button the well-known flip rules pop-up would appear.

![image](https://user-images.githubusercontent.com/38664167/221896265-1aa115df-b2ac-4df9-bc47-b6a5130bd9c4.png)

There the Reporter needs to select the rule that is being broken. In the pop-up "Report" button will close the pop-up and set this rule as a report reason. Clicking "Back" button would close the pop-up and switch the status to "abstained". As no new UI elements are needed, the process of implementation should be streamlined.

**Comment by m3089237971** [comment]
The keywords given by the wallet are too scarce, and the two keywords lack imagination. The four pictures can't constitute a complete story at all.

**Comment by NPC69** [comment]
I like this proposal because of two reasons:

1. It's useful.
2. It makes bus angry.

**Reply by Noobmasser** [reply]
I believe their concern was extra clicks required during the long session. I would say that if you get around 8 reports, 16 more mouse clicks than usual are a fair price for building a better, more sustainable community. People are complaining about flips quality, but this feature would allow the ones who failed to learn on their mistakes, gradually increasing overall flip quality.

**Reply by NPC69** [reply]
Extra clicks? Oh, no! Anyway...
https://m.media-amazon.com/images/I/81t6A8K7jTL.png

**Reply by krairy** [reply]
Well, if you know why you report someone's flip, these 16 clicks will not cost you much time. 
If giving the reason of reporting takes too much time, you'd better think twice. 

One reasons may be given some priority over others to make the choice easier..?

**Comment by m3089237971** [comment]
Wallet needs to provide more keyword groups and imaginative keywords, which is the key to flipping.

**Comment by krairy** [comment]
So well formulated. Especially the part with invalidating reports when no consensus achieved. 

Can't imagine why this has not been implemented yet.

**Reply by Noobmasser** [reply]
Hello and thanks for the input. I edited the post and made a clarification that the part about invalidating reports if the reason consensus has not been reached **is not a part of the current proposal**. As @cryptogeek555 noted in a separate thread, it might lead to invalidating report of the flip that breaks several rules at once.

**Comment by cryptogeek555** [comment]
i like the idea, but what if a flip doesn't follow several rules? in that case those flips will not be reported as some people will mark one reason, other people will mark the other reason and there will be no consensus.

**Reply by krairy** [reply]
Legit concern. What about ranging rules by priority?

**Reply by Noobmasser** [reply]
I edited the post and made a clarification that the part about invalidating reports if the reason consensus has not been reached **is not a part of the current proposal**. I don't think it needs to be discussed in the current scope. The current proposal is purely cosmetic from the logic standpoint, but should have concrete repercussions and lead to better understanding overall.

**Reply by cryptogeek555** [reply]
i'm afraid i don't agree with you here. This should be clearly addressed in the proposal because it is the most tricky part. If we do not address it, people just take advantage of it and start making those kind of flips (for example, waking up, painter and the last one a painting with two keywords on it and all of them will have numbers 1,2,3,4 in corners to mislead what reason to choose).  And we will be flooded with bad flips again...

**Reply by krairy** [reply]
I guess we are talking about two aspects here: 
1) Giving interface for providing the reasons of report
2) Collecting the reasons and checking the consensus 

Implementation of at least the 1st one will make me happy

**Reply by NPC69** [reply]
Then people should be allowed to select multiple reasons, which can further strengthen the validity of consensus. If people don't agree on the combination of the reasons for reporting then a consensus is not reached.

**Comment by wji** [comment]
I think something like this (the interface part) could be useful but depends on how it's implemented. The way the reporting currently works to my knowledge is by setting the byte for the grade in the long answer tx. I wouldn't make changes to the transactions just for providing this guidance to the flip makers. 

I think it's sometimes a little ambiguous what constitutes a bad flip in the end. It could be effectively anything that can be easily figured out with ML or that is too confusing for the average human. The rules shown on the interface have changed over time and if this was an additional field in the txs it would also have to be changed each time.

**Comment by 0JustinMiles0** [comment]
Any change that asks for more user inputs during the validation session should add a substantive benefit. I think it would be a nice to have but I don't think it would bring a substantive benefit hence I'm not in favor of this idea.

**Comment by NPC69** [comment]
We urgently need this or something like this. The situation is getting out of hand.

**Comment by krairy** [comment]
Oh. No rule selection yet - how is the idena project still alive??

**Reply by NPC69** [reply]
I don't even know. I guess we got a resuscitation injection from Gitcoin. But having this problem fixed we will be more than just solid.
